include/picongpu/particles/collision/InterCollision.hpp

```diff
@@ -146,9 +145,11 @@ namespace picongpu
                     // offset of the superCell (in cells, without any guards) to the
                     // origin of the local domain
                     DataSpace<simDim> const localSuperCellOffset = superCellIdx - mapper.getGuardingSuperCells();
-                    rngHandle.init(
-                        localSuperCellOffset * SuperCellSize::toRT()
-                        + DataSpaceOperations<simDim>::template map<SuperCellSize>(worker.getWorkerIdx()));
+                    auto rngOffset = DataSpace<simDim>::create(0);
+                    rngOffset.x() = worker.getWorkerIdx();
+                    auto numRNGsPerSuperCell = DataSpace<simDim>::create(1);
+                    numRNGsPerSuperCell.x() = numFrameSlots;
+                    rngHandle.init(localSuperCellOffset * numRNGsPerSuperCell + rngOffset);
```

Comment:
The storage format of the RNG states changed. In the past, we used one RNG state per cell. This is changes within this PR where we have a linear array of RNG states per superCell but stored in an array [x-superCells*RNG-states-perSuperCell,y-supercell,z-supercell].

This will be implemented in the future a little bit more cleanly so that we really work with a hierarchy supercell equal to the particle hierarchy structure.